### PR TITLE
unnecessary conditions

### PR DESCRIPTION
**Pull request by danielrohers**

### CODE REVIEW

File: lib/query.js [hunk 1]
```diff
@@ -1127,9 +1123,7 @@ Query.prototype.findOne = function (conditions, fields, options, callback) {
 
   if (options) {
     this.setOptions(options);
-  }
-
-  if (fields) {
+  } else if (fields) {
```
Comment (vkarpov15) [inline lib/query.js]: This isn't necessarily correct, both fields and options can be defined. You're right about the other two ifs being redundant though. If you can get rid of this change I can merge this PR

Comment (danielrohers) [inline lib/query.js]: Really this case I'm wrong. But the other two ifs are redundant.

Comment (vkarpov15) [inline lib/query.js]: Yeah the other two ifs are redundant, you're right. Just remove the L-1126 change and this is good :)

**Comment by danielrohers** [comment]
@vkarpov15 ready!!

**Pull request merged**